src/ModelObjects/representations.rs

```diff
@@ -106,62 +106,59 @@ impl<'a> SystemRepresentation {
         }
     }
 
-    pub fn collect_open_inputs(
+    pub fn collect_inputs_edges(
         &'a self,
         states: &[State<'a>],
         action: &str,
-    ) -> Result<Vec<(&'a Component, Vec<&'a Edge>, usize)>, String> {
+    ) -> Vec<(&'a Component, Vec<&'a Edge>, usize)> {
         let mut edges = vec![];
         let mut index = 0;
 
-        if self.collect_open_edges(states, &mut index, action, &mut edges, &SyncType::Input) {
-            Ok(edges)
-        } else {
-            Err("Conjunction rules on output not satisfied".to_string())
-        }
+        self.collect_edges(states, &mut index, action, &mut edges, &SyncType::Input);
+
+        edges
     }
 
-    pub fn collect_open_outputs(
+    pub fn collect_outputs_edges(
         &'a self,
         states: &[State<'a>],
         action: &str,
-    ) -> Result<Vec<(&'a Component, Vec<&'a Edge>, usize)>, String> {
+    ) -> Vec<(&'a Component, Vec<&'a Edge>, usize)> {
         let mut edges = vec![];
         let mut index = 0;
 
-        if self.collect_open_edges(states, &mut index, action, &mut edges, &SyncType::Output) {
-            Ok(edges)
-        } else {
-            Err("Conjunction rules on input not satisfied".to_string())
-        }
+        self.collect_edges(states, &mut index, action, &mut edges, &SyncType::Output);
+
+        edges
     }
 
-    fn collect_open_edges(
+    fn collect_edges(
         &'a self,
         states: &[State<'a>],
         index: &mut usize,
         action: &str,
         open_edges: &mut Vec<(&'a Component, Vec<&'a Edge>, usize)>,
         sync_type: &SyncType,
-    ) -> bool {
+    ) {
         match self {
             SystemRepresentation::Composition(left_side, right_side) => {
-                left_side.collect_open_edges(states, index, action, open_edges, sync_type)
-                    || right_side.collect_open_edges(states, index, action, open_edges, sync_type)
+                left_side.collect_edges(states, index, action, open_edges, sync_type);
+                right_side.collect_edges(states, index, action, open_edges, sync_type);
```

Comment:
The test `tests::refinement::AG_Tests::AG_Tests::AImpNotRefinesAG` is fixed as the short circuiting here is removed.
When short circuiting happend the `right_side ` edges would not be added to `open_edges`